packages/cloud_firestore/lib/src/set_options.dart

```diff
@@ -0,0 +1,21 @@
+// Copyright 2017, the Chromium project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+part of cloud_firestore;
+
+/// An options object that configures the behavior of [setData()] calls.
+///
+/// By providing the [SetOptions] objects returned by [merge], the [setData()]
+/// calls on [DocumentReference] can be configured to perform granular merges
+/// instead of overwriting the target documents in their entirety.
+class SetOptions {
```

Comment:
The Android Firestore SDKs seem to provides more advanced `SetOptions` functionality, but it doesn't seem to be available yet on iOS. For future compatibility I'm making this a class instead of the more Dart-y optional named arguments.